cylc/flow/run_modes/skip.py

```diff
@@ -0,0 +1,166 @@
+    itask: 'TaskProxy',
+    now: Tuple[float, str]
+) -> 'Literal[True]':
+    """Submit a task in skip mode.
+
+    Returns:
+        True - indicating that TaskJobManager need take no further action.
+    """
+    task_job_mgr._set_retry_timers(itask, itask.tdef.rtconfig)
+    itask.summary['started_time'] = now[0]
+    itask.waiting_on_job_prep = False
+    itask.submit_num += 1
+
+    itask.platform = get_platform()
+    itask.platform['name'] = RunMode.SKIP
+    itask.summary['job_runner_name'] = RunMode.SKIP
+    itask.tdef.run_mode = RunMode.SKIP
+    task_job_mgr.task_events_mgr.process_message(
+        itask, INFO, TASK_OUTPUT_SUBMITTED,
+    )
+    task_job_mgr.workflow_db_mgr.put_insert_task_jobs(
+        itask, {
+            'time_submit': now[1],
+            'try_num': itask.get_try_num(),
+            'flow_nums': str(list(itask.flow_nums)),
+            'is_manual_submit': itask.is_manual_submit,
+            'job_runner_name': RunMode.SIMULATION,
+            'platform_name': RunMode.SIMULATION,
+            'submit_status': 0   # Submission has succeeded
+        }
+    )
+    for output in process_outputs(itask):
+        task_job_mgr.task_events_mgr.process_message(itask, INFO, output)
+
+    return True
+
+
+def process_outputs(itask: 'TaskProxy') -> List[str]:
+    """Process Skip Mode Outputs:
+
+    * By default, all required outputs will be generated plus succeeded
+      if success is optional.
+    * The outputs submitted and started are always produced and do not
+      need to be defined in outputs.
+    * If outputs is specified and does not include either
+      succeeded or failed then succeeded will be produced.
+
+    Return:
+        A list of outputs to emit.
+    """
+    result: List[str] = []
+    conf_outputs = itask.tdef.rtconfig['skip']['outputs']
+
+    # Remove started or submitted from our list of outputs:
+    for out in get_unecessary_outputs(conf_outputs):
+        conf_outputs.remove(out)
+
+    # Always produce `submitted` output:
+    result.append(TASK_OUTPUT_SUBMITTED)
+    # (No need to produce `started` as this is automatically handled by
+    # task event manager for jobless modes)
+
+    # Send the rest of our outputs, unless they are succeed or failed,
+    # which we hold back, to prevent warnings about pre-requisites being
+    # unmet being shown because a "finished" output happens to come first.
+    for message in itask.state.outputs.iter_required_messages():
+        trigger = itask.state.outputs._message_to_trigger[message]
+        # Send message unless it be succeeded/failed.
+        if trigger in [TASK_OUTPUT_SUCCEEDED, TASK_OUTPUT_FAILED]:
+            continue
+        if not conf_outputs or trigger in conf_outputs:
+            result.append(message)
+
+    # Send succeeded/failed last.
+    if TASK_OUTPUT_FAILED in conf_outputs:
+        result.append(TASK_OUTPUT_FAILED)
+    else:
+        result.append(TASK_OUTPUT_SUCCEEDED)
+
+    return result
```

Comment:
The `conf_outputs.remove` was directly modifying the `TaskDef` object which will change the task configuration for future submissions. This may highlight a test hole, might be worth adding a test which runs the same task in subsequent cycles?

I've also removed a `result.append` statement for `TASK_OUTPUT_SUBMITTED` (as this output was always added) and switched from `in []` to `in {}` for marginal performance gain.

```suggestion
    # Always produce `submitted` output:
    # (No need to produce `started` as this is automatically handled by
    # task event manager for jobless modes)
    result: List[str] = [TASK_OUTPUT_SUBMITTED]
    
    conf_outputs = list(itask.tdef.rtconfig['skip']['outputs'])

    # Remove started or submitted from our list of outputs:
    for out in get_unecessary_outputs(conf_outputs):
        conf_outputs.remove(out)

    # Send the rest of our outputs, unless they are succeed or failed,
    # which we hold back, to prevent warnings about pre-requisites being
    # unmet being shown because a "finished" output happens to come first.
    for message in itask.state.outputs.iter_required_messages():
        trigger = itask.state.outputs._message_to_trigger[message]
        # Send message unless it be succeeded/failed.
        if (
            trigger not in {TASK_OUTPUT_SUCCEEDED, TASK_OUTPUT_FAILED}:
            and not conf_outputs or trigger in conf_outputs
        ):
            result.append(message)

    # Send succeeded/failed last.
    if TASK_OUTPUT_FAILED in conf_outputs:
        result.append(TASK_OUTPUT_FAILED)
    else:
        result.append(TASK_OUTPUT_SUCCEEDED)

    return result
```